core/resource/service.go

```diff
@@ -67,32 +67,40 @@ type NamespaceService interface {
 	List(ctx context.Context) ([]namespace.Namespace, error)
 }
 
+type SchemaService interface {
+	RunMigrations(ctx context.Context) error
+}
+
 type Service struct {
 	logger              log.Logger
+	appConfig           AppConfig
 	repository          Repository
-	configRepository    ConfigRepository
+	schemaRepository    SchemaRepository
 	relationService     RelationService
 	userService         UserService
 	projectService      ProjectService
 	organizationService OrganizationService
 	groupService        GroupService
 	policyService       PolicyService
 	namespaceService    NamespaceService
+	schemaService       SchemaService
```

Comment:
why do we have both schema repository and schema service as dependencies here?
shouldn't we just depend on schema service and the operation would be on schema service, instead of resource service depend on schema repository?

Reply:
That would be more ideal. So the idea behind this was because we need to keep enabling the upsert API to store to db even though resource path is configured to file system (for migration). So I was thinking to keep schema service to handle schema related logic only and leave API related logic in the core.